cpp/core/jni/JniWrapper.cc

```diff
@@ -770,6 +779,87 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleSplitterJniWrapp
+    jlong firstBatchHandle) {
+  JNI_METHOD_START
+  if (partitioning_name_jstr == NULL) {
+    gluten::JniThrow(std::string("Short partitioning name can't be null"));
+    return 0;
+  }
+  auto partitioning_name_c = env->GetStringUTFChars(partitioning_name_jstr, JNI_FALSE);
+  auto partitioning_name = std::string(partitioning_name_c);
+  env->ReleaseStringUTFChars(partitioning_name_jstr, partitioning_name_c);
+  auto splitOptions = SplitOptions::Defaults();
+  splitOptions.buffered_write = true;
+  splitOptions.is_celeborn = true;
+  if (buffer_size > 0) {
+    splitOptions.buffer_size = buffer_size;
+  }
+  if (push_buffer_max_size > 0) {
+    splitOptions.push_buffer_max_size = push_buffer_max_size;
+  }
+  splitOptions.offheap_per_task = offheap_per_task;
+  if (compression_type_jstr != NULL) {
+    auto compression_type_result = GetCompressionType(env, compression_type_jstr);
+    if (compression_type_result.status().ok()) {
+      splitOptions.compression_type = compression_type_result.MoveValueUnsafe();
+    }
+  }
+
+  auto* allocator = reinterpret_cast<MemoryAllocator*>(allocator_id);
+  if (allocator == nullptr) {
+    gluten::JniThrow("Memory pool does not exist or has been closed");
+  }
+  splitOptions.memory_pool = AsWrappedArrowMemoryPool(allocator);
+  jclass cls = env->FindClass("java/lang/Thread");
+  jmethodID mid = env->GetStaticMethodID(cls, "currentThread", "()Ljava/lang/Thread;");
+  jobject thread = env->CallStaticObjectMethod(cls, mid);
+  if (thread == NULL) {
+    std::cout << "Thread.currentThread() return NULL" << std::endl;
+  } else {
+    jmethodID mid_getid = GetMethodIDOrError(env, cls, "getId", "()J");
+    jlong sid = env->CallLongMethod(thread, mid_getid);
+    splitOptions.thread_id = (int64_t)sid;
+  }
+  jclass tc_cls = env->FindClass("org/apache/spark/TaskContext");
+  jmethodID get_tc_mid = env->GetStaticMethodID(tc_cls, "get", "()Lorg/apache/spark/TaskContext;");
+  jobject tc_obj = env->CallStaticObjectMethod(tc_cls, get_tc_mid);
+  if (tc_obj == NULL) {
+    std::cout << "TaskContext.get() return NULL" << std::endl;
+  } else {
+    jmethodID get_tsk_attmpt_mid = GetMethodIDOrError(env, tc_cls, "taskAttemptId", "()J");
+    jlong attmpt_id = env->CallLongMethod(tc_obj, get_tsk_attmpt_mid);
+    splitOptions.task_attempt_id = (int64_t)attmpt_id;
+  }
+  splitOptions.batch_compress_threshold = batch_compress_threshold;
+  JavaVM* vm;
+  if (env->GetJavaVM(&vm) != JNI_OK) {
+    gluten::JniThrow("Unable to get JavaVM instance");
+  }
+  std::shared_ptr<CelebornClient> celeborn_client =
+      std::make_shared<CelebornClient>(vm, celeborn_partition_pusher, celeborn_push_partition_data_method);
+
+  splitOptions.celeborn_client = std::move(celeborn_client);
+
+  auto backend = gluten::CreateBackend();
+  auto batch = gluten_columnarbatch_holder_.Lookup(firstBatchHandle);
+  auto splitter = backend->makeSplitter(partitioning_name, num_partitions, std::move(splitOptions), batch->GetType());
```

Comment:
For all the backend, we will use `CelebornSplitter` here, so we do not need to use backend->makeSplitter

Reply:
OK

Reply:
Let's note in future we should take some refactors to merge the JNI functions for making different types of splitters.

So in this PR it's OK to keep the code as it is.

Reply:
@zhztheplayer ok, is there anything else that needs to be optimized?

Reply:
Let's merge this since it's in good shape. And let's raise more PRs to improve splitter API's extensibility.